src/ast/passes/semantic_analyser.cpp

```diff
@@ -1529,30 +1529,71 @@ Probe *SemanticAnalyser::get_probe_from_scope(Scope *scope,
 
 void SemanticAnalyser::visit(Map &map)
 {
-  MapKey key;
+  SizedType key = CreateNone();
+
+  if (map.key_expr) {
+    Visit(map.key_expr);
+    Expression *expr = map.key_expr;
+
+    if (expr->type.IsTupleTy()) {
+      // We can only do this expression and size type manipulation
+      // if we have access to the underlying expressions in the tuple ast
+      // e.g. @map[(1, 2)] vs $a = (1, 2) @map[$a]
+      auto tuple_ast = dynamic_cast<Tuple *>(expr);
```

Comment:
What's this manipulation needed for?

If it's necessary, isn't it necessary for non-literal tuples as well? If it's not, then can this code just be deleted?

Reply:
This is to maintain backward compat with the current multi-key implementation where we insert casts and increase the size of ints to 64 bits. We can't do this for non-literal tuples (as the comment says) because we don't have access to the underlying expressions for non-literal tuples.

Reply:
Related: https://github.com/bpftrace/bpftrace/issues/3439

Reply:
Now that #3443 has been merged (and might have fixed #3439), is this code needed any more?

Reply:
I tried removing it and some tests are still failing e.g. `Casting retval should work`. Mind if I work on fixing this in a follow-up PR? It might require extended investigation.

Reply:
Nevermind. I figured it out and removed the casting.